phoenicis-repository/src/main/java/org/phoenicis/repository/types/GitRepository.java

```diff
@@ -65,72 +69,103 @@ private File createRepositoryLocation(String cacheDirectoryPath) {
         return new File(cacheDirectoryPath + "/git" + hashcode);
     }
 
-    @Override
-    public synchronized RepositoryDTO fetchInstallableApplications() {
-        RepositoryDTO result = null;
-        try {
-            mutex.acquire();
+    private void cloneOrUpdate() throws RepositoryException {
+        synchronized (mutex) {
```

Comment:
So basically we can only clone **one** git repository at a time, independent on the repository we want to clone, right?

Reply:
Yes. I don't think this limitation is important for us.

Reply:
I'm not sure.
I think it would be useful to be able to fetch multiple different repositories at the same time.
This is only possible if we use a different mutex for each repository location.

Reply:
I don't currently see how this could be implemented. Apart from that, I'm not really sure it's worth the effort because it's nothing that will normally happen/have much of an effect.

Reply:
It is ok for now, it is just possible that we need to revise it later on if fetching multiple repositories takes to long.